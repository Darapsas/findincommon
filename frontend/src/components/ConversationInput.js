import React, { Fragment } from "react";
import { Formik, Form, Field } from "formik";
import * as Yup from "yup";

const messageSchema = Yup.object().shape({
  /*name: Yup.string().required("This is a required field"),
  description: Yup.string().max(
    250,
    "Description must be at most 250 characters long."
  ),
  startDate: Yup.date()
    .min(new Date(), "You cannot create an message for the past.")
    .required("Start date and time are required"),
  endDate: Yup.date()
    .min(new Date(), "You cannot create an message for the past.")
    .required("End date and time are required")*/
});

export default props => {
  return (
    <Formik
      initialValues={{
        creatorId: "", //props.participant.id,
        creatorFirstName: "", //props.participant.firstName,
        creatorLastName: "", //props.participant.lastName,
        text: ""
      }}
      validationSchema={messageSchema}
      //  validate={}
      onSubmit={(values, actions) => {
        fetch(`http://192.168.99.100:8080/api/messages`, {
          method: "POST",
          body: JSON.stringify(values),
          headers: {
            "Content-Type": "application/json"
          }
        })
          .then(response => {
            console.log("Successfully created", JSON.stringify(response));
            //props.history.goBack();
            values.isSubmitting = false;
          })
          .catch(error => console.error("Error:", error));
      }}
      render={({ errors, touched, isSubmitting, values }) => (
        <Form>
          <div className="form-row">
            <div className="form-group col-sm-10">
              <Field
                className="form-control"
                name="text"
                type="text"
                placeholder="Write your message here"
              />
              {touched.text && errors.text && (
                <small className="form-text text-muted alert alert-danger">
                  {errors.text}
                </small>
              )}
            </div>

            <div className="form-group col-sm-2">
              <button
                className="btn btn-primary"
                type="submit"
                disabled={isSubmitting}
              >
                Send
              </button>
            </div>
          </div>
        </Form>
      )}
    />
  );
};
