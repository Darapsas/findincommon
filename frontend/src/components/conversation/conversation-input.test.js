import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import ConversationInput from './conversation-input'

describe('<ConversationInput />', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<ConversationInput />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
