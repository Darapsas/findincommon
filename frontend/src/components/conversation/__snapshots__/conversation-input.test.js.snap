// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConversationInput /> matches the snapshot 1`] = `
<form
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="form-row"
  >
    <div
      className="form-group col-sm-10"
    >
      <input
        className="form-control"
        name="text"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Write your message here"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group col-sm-2"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        type="submit"
      >
        Send
      </button>
    </div>
  </div>
</form>
`;
